scripts/ci/ensure-all-node-requirements-available.sh

```diff
@@ -91,7 +91,7 @@ git ls-tree -r HEAD | while IFS='' read line; do
             done
           fi
           # Check optionalDependencies from package.json
-          if [[ $requirement_found -eq 0 ]]; then
+          if [[ $requirement_found -eq 0 ]] && [[ -n "${NPM_OPTIONAL_MODULES:-}" ]]; then
```

Comment:
Small nit, but I think I'd rather revert this line to how it was, and then use http://stackoverflow.com/questions/7577052/bash-empty-array-expansion-with-set-u to change the line below to:
```
for module in "${NPM_OPTIONAL_MODULES[@]+${NPM_OPTIONAL_MODULES[@]}}"; do
```
(and as with my earlier comment, for extra robustness similar changes could be applied to lines 86 and 104)